src/TerminalFighter/MainMenu/MainMenu.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <chrono>
+#include <unistd.h>
+
+#include <SDL_ttf.h>
+#include <SDL2/SDL.h>
+
+#include "../TextArea/TextArea.h"
+
+
+class MainMenu {
+public:
+    MainMenu(std::string font_path, int width, int height);
+    ~MainMenu();
+    enum Options { START, QUIT };
+    void draw(SDL_Renderer& renderer);
+    void move_up_selection();
+    void move_down_selection();
+    void update();
+    MainMenu::Options get_current_selection();
+
+private:
+    TextArea title_;
```

Comment:
These should use the I_TextArea interface. We should probably use Dependency injection so we can instantiate this class with fake text areas.

Reply:
Actually, fuck the DI, just make them use the interface class. This one can't be because it has to be instantiated, but you can do it for the vector of text  areas.

Reply:
I dont get it, doesn't TextArea use the I_TextArea interface already?

Reply:
The important part is that stuff using TextArea uses I_TextArea, this way if we create a class that depends on I_TextArea, we don't have to pass in an actual TextArea, we can pass in a MockTestArea (testing purposes!)

Reply:
@ctlai95 , do you understand this^ We can re-phrase if needed

Reply:
@bseto worked on it wtih enoch